src/reducers/fileTree.tsx

```diff
@@ -157,6 +159,74 @@ export const buildFileTree = (version: Version): FileTree => {
   };
 };
 
+type GetRelativePathParams = {
+  _log: typeof log;
+  currentPath: string;
+  pathList: string[];
+  position: 'previous' | 'next';
+};
+
+export const getRelativePath = ({
+  _log = log,
+  currentPath,
+  pathList,
+  position,
+}: GetRelativePathParams): string | void => {
+  const currentIndex = pathList.indexOf(currentPath);
+  if (!currentIndex) {
+    _log.debug(`Cannot find ${currentPath} in pathList: ${pathList}`);
+    return undefined;
+  }
+  const newIndex =
+    position === 'previous' ? currentIndex - 1 : currentIndex + 1;
+  const validIndex = newIndex >= 0 && pathList.length >= newIndex;
+  if (!validIndex) {
+    _log.debug(`There is not a ${position} file wrt ${currentPath}`);
+  }
+  return validIndex ? pathList[newIndex] : undefined;
+};
+
+type GoToNextFileParams = {
+  _log: typeof log;
+  currentPath: string;
+};
+
+export const goToNextFile = ({
+  _log = log,
+  currentPath,
+}: GoToNextFileParams): ThunkActionCreator => {
+  return async (dispatch, getState) => {
+    const { fileTree: fileTreeState } = getState();
+    const { forVersionId, tree } = fileTreeState;
+    const { pathList } = tree;
+
+    if (!forVersionId || !pathList) {
+      _log.debug('Cannot navigate to next file with no version loaded');
+      return;
+    }
+
+    const nextPath = getRelativePath({
+      _log,
+      currentPath,
+      pathList,
+      position: 'next',
+    });
+
+    if (!nextPath) {
+      // There is no next file, so do nothing.
+      // TODO: I don't think we need to log in this case.
+      return;
```

Comment:
If I'm pressing a key to see the next message and nothing happens, I'd be pretty confused. You could dispatch an action here that shows a message like 'There are no more linter messages'. Because of this, I think using a thunk for this feature is the right way to go.

Reply:
Couldn't we loop over the files/messages?

Reply:
> Couldn't we loop over the files/messages?

This is a good question. Do we think it would make sense to go to the first file if a user chooses "next file" when we're at the last file? The same question for "previous file" - should that go to the last file if the user is on the first file? In some ways this makes sense, but in other ways it does not. I think I'm leaning towards "yes" though.

@wagnerand What do you think?

Reply:
FYI, for now I will implement @willdurand's suggestion as it saves us from having to inform the user if there are no more files in the direction they wish to navigate.

Reply:
> If I'm pressing a key to see the next message and nothing happens, I'd be pretty confused. 
>> Couldn't we loop over the files/messages?

I considered this. I find this type of behavior (which is common for a find/replace UX) very confusing because with a long list of results, it's hard to know when you're *finished*. In this case, navigating through messages is a task that has a finite end state. We could probably address this problem in another way, though. For example, maybe the first item in the list has some kind of animation to give you a cue.